modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaConnectorTelemetryOverride.php

```diff
@@ -26,8 +38,120 @@ public function onTerminateResponse(KernelEvent $event) {
     // Check if telemetry opted,
     // then only trigger AcquiaTelemetry::onTerminateResponse($event).
     if ($this->state->get('acquia_connector.telemetry.opted', TRUE)) {
+      $this->sendTelemetry("ACMS Telemetry data");
       parent::onTerminateResponse($event);
     }
   }
 
+  /**
+   * Returns the Amplitude API key.
+   *
+   * This is not intended to be private. It is typically included in client
+   * side code. Fetching data requires an additional API secret.
+   *
+   * @see https://developers.amplitude.com/#http-api
+   *
+   * @return string
+   *   The Amplitude API key.
+   */
+  private function getApiKey() {
+    return Settings::get('acquia_connector.telemetry.key', 'somedummy key');
+  }
+
+  /**
+   * Sends an event to Amplitude.
+   *
+   * @param array $event
+   *   The Amplitude event.
+   *
+   * @throws \GuzzleHttp\Exception\GuzzleException
+   *
+   * @see https://developers.amplitude.com/#http-api
+   */
+  private function sendEvent(array $event) {
+    $this->httpClient->request('POST', $this->apiUrl, [
+      'form_params' => [
+        'api_key' => $this->getApiKey(),
+        'event' => Json::encode($event),
+      ],
+    ]);
+  }
+
+  /**
+   * Creates and sends an event to Amplitude.
+   *
+   * @param string $event_type
+   *   The event type. This accepts any string that is not reserved. Reserved
+   *   event types include: "[Amplitude] Start Session", "[Amplitude] End
+   *   Session", "[Amplitude] Revenue", "[Amplitude] Revenue (Verified)",
+   *   "[Amplitude] Revenue (Unverified)", and "[Amplitude] Merged User".
+   * @param array $event_properties
+   *   (optional) Event properties.
+   *
+   * @return bool
+   *   TRUE if event was successfully sent, otherwise FALSE.
+   *
+   * @throws \Exception
+   *   Thrown if state key acquia_telemetry.loud is TRUE and request fails.
+   *
+   * @see https://amplitude.zendesk.com/hc/en-us/articles/204771828#keys-for-the-event-argument
+   */
+  public function sendTelemetry($event_type, array $event_properties = []) {
```

Comment:
Return statement is missing

Reply:
Add data type for parameter $event_type

Reply:
$event_type is string type , i dont think we need to mention that